df2gspread/df2gspread.py

```diff
@@ -82,12 +104,18 @@ def upload(df, gfile="/New Spreadsheet", wks_name=None, chunk_size=1000,
         last_col = ascii_uppercase[seq_num % len(ascii_uppercase)] + last_col
         seq_num = seq_num // len(ascii_uppercase) - 1
 
-    # if pandas df large then given worksheet then increes num of cols or rows
-    if len(df.index) + 1 > wks.row_count:
-        wks.add_rows(len(df.index) - wks.row_count + 1)
-
-    if len(df.columns) + 1 > wks.col_count:
-        wks.add_cols(len(df.columns) - wks.col_count + 1)
+    # If user requested to resize sheet to fit dataframe, go ahead and 
+    # resize larger or smaller to better match new size of pandas dataframe.
+    # Otherwise, leave it the same size unless the sheet needs to be expanded
+    # to accomodate a larger dataframe.
+    extra_col = 1 if col_names else 0
+    extra_row = 1 if row_names else 0
+    if df_size:
+        wks.resize(rows=len(df.index) + extra_row, cols=len(df.columns) + extra_col)
```

Comment:
`len(df.index) + row_names` gives the same result as `len(df.index) + extra_row`..
I'm not sure that `extra_row = 1 if row_names else 0` adds more sense.. Would it be better to get rid of these two lines?